File: 633. Sum of Square Numbers/squaresum.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a = 0  # Initialize variable a to 0
        while a * a <= c:  # Loop until a^2 exceeds or equals c
            b = int((c - a * a) ** 0.5)  # Calculate the square root of (c - a^2) and convert it to an integer
            if b * b + a * a == c:  # Check if b^2 + a^2 equals c
                return True  # If a solution is found, return True
            a += 1  # Increment a by 1 for the next iteration
        return False  # If no solution is found, return False

File: 633. Sum of Square Numbers/squaresum.c
bool judgeSquareSum(int c){
    if (c < 0)
        return false;

    int a = 0; // Initialize a to 0
    int b = (int) sqrt(c); // Initialize b as square root of c

    while (a <= b) { // Loop until a exceeds b
        long long sumOfSquares = (long long) a * a + (long long) b * b;; // Calculate the sum of squares
        
        if (sumOfSquares == c)
            return true; // If the sum of squares is equal to c, return true
        else if (sumOfSquares < c)
            a++; // If the sum of squares is less than c, increment a
        else
            b--; // If the sum of squares is greater than c, decrement b
    }

    return false; // If no solution is found, return false
}

